crates/binjs_io/src/lib.rs

```diff
@@ -414,14 +414,16 @@ impl Format {
     /// `matches` as a subcommand. If none, pick the default
     /// provider, without any command-line arguments.
     pub fn from_matches(matches: &clap::ArgMatches) -> Result<Self, std::io::Error> {
-        for provider in Self::providers().into_iter() {
-            let subcommand = provider.subcommand();
-            let key = subcommand.get_name();
-            if let Some(matches) = matches.subcommand_matches(key) {
-                return provider.handle_subcommand(Some(matches));
+        if let Some(matches) = matches.subcommand_matches("advanced") {
```

Comment:
Mmmh... this puts half of `advanced` in encode.rs and half here.

Would it be possible to also make `fn providers()` private and replace it with a `pub fn subcommand() -> SubCommand` that would specify `"advanced"` from this file?